kubernetes/grpc-bookstore.yaml

```diff
@@ -31,44 +31,47 @@ spec:
     app: esp-grpc-bookstore
   type: LoadBalancer
 ---
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
 kind: Deployment
 metadata:
   name: esp-grpc-bookstore
 spec:
   replicas: 1
+  selector:
+    matchLabels:
+      app: esp-grpc-bookstore
   template:
     metadata:
       labels:
         app: esp-grpc-bookstore
     spec:
       # [START secret-1]
       volumes:
-        - name: service-account-creds
-          secret:
-            secretName: service-account-creds
+      - name: service-account-creds
+        secret:
+          secretName: service-account-creds
       # [END secret-1]
       # [START service]
       containers:
-        - name: esp
-          image: gcr.io/endpoints-release/endpoints-runtime:2
-          args: [
-            "--listener_port=9000",
-            "--service=SERVICE_NAME",
-            "--rollout_strategy=managed",
-            "--backend=grpc://127.0.0.1:8000",
-            "--service_account_key=/etc/esp/creds/service-account-creds.json",
-          ]
-      # [END service]
+      - name: esp
+        image: gcr.io/endpoints-release/endpoints-runtime:2
+        args: [
+          "--listener_port=9000",
+          "--service=SERVICE_NAME",
+          "--rollout_strategy=managed",
+          "--backend=grpc://127.0.0.1:8000",
+          "--service_account_key=/etc/esp/creds/service-account-creds.json",
+        ]
+        # [END service]
```

Comment:
Remove this START/END service?

Reply:
No, there are used in the doc

Reply:
I have gone through the k8s doc and haven't seen any place mentioning this [START/END service]. Maybe [here](https://cloud.devsite.corp.google.com/endpoints/docs/grpc/get-started-kubernetes#configuring_the_service_name_and_starting_the_service)?

Not sure about how they work for doc but they look confusing in this yaml.

Reply:
That is an example of where they are used. It's needed to show a subsection. I don't think it's too confusing, it's commented out.

Reply:
If this yaml stands alone, this subsection definitely doesn't define a service and if they do, it should include the whole block under `container`.

Maybe change it to be START/END esp

Reply:
Not, we could not change its name.  It is used in the doc.  Here is the context

{% include "cloud/_docwidgets/_github_include.html" with project="endpoints-samples" region_tag="service" file="kubernetes/grpc-bookstore.yaml" prettyprint_lang="yaml" %}

In its region_tag

Reply:
ACK